dev/ColorPicker/ColorPicker.cpp

```diff
@@ -346,13 +346,14 @@ void ColorPicker::OnIsAlphaEnabledChanged(winrt::DependencyPropertyChangedEventA
 {
     m_currentHex = GetCurrentHexValue();
 
-    if (m_hexTextBox)
+    if (auto hexTextBox = m_hexTextBox.get())
```

Comment:
auto&&

Reply:
Interesting. What is the goal in adding that? Is it to avoid an addref/release?

---
In reply to: [401108450](https://github.com/microsoft/microsoft-ui-xaml/pull/2191#discussion_r401108450) [](ancestors = 401108450)

Reply:
auto&& gets the object exactly as is, accepting any initializer regardless of lvalue or rvalue expression. In addition it preserves const-ness of the original value.  about a year ago Jevan and I did a little investigation into the perf of the different ways of getting a tracker ref variable and auto&& was the most performant.  There is a little bit of risk, if it is possible for you local reference to outlive the the variable tracked by the trackerref you would not want to use auto&& but in most (all?) of our cases that is not possible.

---
In reply to: [401146016](https://github.com/microsoft/microsoft-ui-xaml/pull/2191#discussion_r401146016) [](ancestors = 401146016,401108450)